builtin/providers/aws/resource_aws_s3_bucket_object.go

```diff
@@ -275,9 +277,14 @@ func resourceAwsS3BucketObjectRead(d *schema.ResourceData, meta interface{}) err
 			Key:    aws.String(key),
 		})
 	if err != nil {
-		return fmt.Errorf("Failed to get object tags (bucket: %s, key: %s): %s", bucket, key, err)
+		// Treat the inability to get object tags in a restricted regions as a
+		// soft error.
+		if !restricted {
+			return fmt.Errorf("Failed to get object tags (bucket: %s, key: %s): %s", bucket, key, err)
```

Comment:
@apparentlymart the problem is that there aren't any tags on govcloud anyway, so if you want to error you probably need to do so on the put, not the get.